src/serialization/sb2.js

```diff
@@ -191,11 +191,13 @@ const parseScratchObject = function (object, runtime, topLevel) {
     if (object.hasOwnProperty('variables')) {
         for (let j = 0; j < object.variables.length; j++) {
             const variable = object.variables[j];
-            target.variables[variable.name] = new Variable(
+            const newVariable = new Variable(
+                variable.id,
```

Comment:
Ah I see, that is why it works. I think we should explicitly use `undefined`, using `variable.id` when we know it is always `undefined` is bound to confuse someone eventually.